pkg/capr/common.go

```diff
@@ -154,13 +157,14 @@ func GetMachineByOwner(machineCache capicontrollers.MachineCache, obj metav1.Obj
 	return nil, ErrNoMachineOwnerRef
 }
 
-// GetKubectlAndKubeconfigPaths returns the corresponding kubectl/kubeconfig paths for a downstream node for the given kubernetes version.
-func GetKubectlAndKubeconfigPaths(kubernetesVersion string) (string, string) {
-	switch GetRuntime(kubernetesVersion) {
+// GetKubectlAndKubeconfigPaths returns the corresponding kubectl/kubeconfig paths for a downstream node for the given
+// RKEControlPlane.
+func GetKubectlAndKubeconfigPaths(controlPlane *rkev1.RKEControlPlane) (string, string) {
+	switch GetRuntime(controlPlane.Spec.KubernetesVersion) {
 	case RuntimeK3S:
 		return K3sKubectlPath, K3sKubeconfigPath
```

Comment:
is it safe to rely on k3s always linking its installed kubectl as `/usr/local/bin/kubectl`? I'm not sure this is future-proof, as we move onto distros that use transactional updates and/or immutable filesystems that will cause changes to /usr/local/bin to be lost on restart.

I think we should instead switch to using something like:
```golang
var (
  K3sRelativeKubectlPath = "data/current/bin/kubectl"
)
// ...
return path.Join(GetDistroDataDir(controlPlane), K3sRelativeKubectlPath`, K3sKubeconfigPath
```